src/analyses/variable-sensitivity/variable_sensitivity_object_factory.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************\
+
+ Module: analyses variable-sensitivity
+
+ Author: Owen Jones, owen.jones@diffblue.com
+
+\*******************************************************************/
+#include "variable_sensitivity_object_factory.h"
+#include "util/namespace.h"
+
+variable_sensitivity_object_factoryt
+  variable_sensitivity_object_factoryt::s_instance;
+
+/*******************************************************************\
+
+Function: variable_sensitivity_object_factoryt::get_abstract_object_type
+
+ Inputs:
+  type - the type of the variable the abstract object is meant to represent
+
+ Outputs: An enum indicating the abstract object type to use.
+
+ Purpose: Decide which abstract object type to use for the variable in question.
+
+\*******************************************************************/
+
+variable_sensitivity_object_factoryt::ABSTRACT_OBJECT_TYPET
+  variable_sensitivity_object_factoryt::get_abstract_object_type(
+  const typet type)
+{
+  ABSTRACT_OBJECT_TYPET abstract_object_type=TWO_VALUE;
+
+  if(type.id()==ID_signedbv || type.id()==ID_unsignedbv ||
+    type.id()==ID_floatbv || type.id()==ID_fixedbv ||
+    type.id()==ID_c_bool || type.id()==ID_bool)
```

Comment:
@tautschnig So I think this should be also primitive types, is there a definitive list or perhaps a function somewhere?